app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -557,5 +563,46 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
     }
   };
 
+  var getProviderOptionsDescription = function() {
+    API.options('/api/providers?ems_type=' + $scope.emsCommonModel.emstype).then(function(response) {
+      $scope.emsCommonModel.provider_options_description = response;
+      if ($scope.emsCommonModel.provider_options_description.hasOwnProperty('advanced_settings')) {
+        var advanced_settings = $scope.emsCommonModel.provider_options_description.advanced_settings.settings
+        for (section in advanced_settings) {
+          if (advanced_settings.hasOwnProperty(section)){
+            $scope.emsCommonModel.advanced_settings = {}
+            updateProviderOptions($scope.emsCommonModel.advanced_settings, advanced_settings[section]);
+          }
+        }
+      }
+      if ($scope.emsCommonModel.provider_options_description.hasOwnProperty('proxy_settings')) {
+        $scope.emsCommonModel.proxy_settings = {}
+        updateProviderOptions($scope.emsCommonModel, $scope.emsCommonModel.provider_options_description.proxy_settings);
+      }
+    });
+  };
+
+  var updateProviderOptions = function(settings_group, options_section){
+    var section_name = $scope.toSnakeCase(options_section.label);
+    settings_group[section_name] = options_section;
+    settings_group[section_name].section_name = section_name;
+    var section = settings_group[section_name].settings;
+    for (option in section){
+      if (section.hasOwnProperty(option)) {
+        if ($scope.emsCommonModel.provider_options_values.hasOwnProperty(section_name) &&
+          $scope.emsCommonModel.provider_options_values[section_name].hasOwnProperty(option)){
+          section[option].value = $scope.emsCommonModel.provider_options_values[section_name][option];
+        } else {
+          section[option].value = "";
+        }
+        section[option].name = option;
+      }
+    }
+  };
+
+  $scope.toSnakeCase = function(st) {
```

Comment:
You can probably use [_.snakeCase()](https://lodash.com/docs/3.10.1#snakeCase) instead..